sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```diff
@@ -177,6 +167,42 @@ case class ExplainCommand(
   }
 }
 
+object ExplainCommand {
+  /**
+   * Initializes an `ExplainCommand` object by passing `LogicalPlan`. This logical plan will be
+   * run through the analyzer and optimizer when this command is actually run.
+   */
+  def apply(
+    logicalPlan: LogicalPlan,
+    extended: Boolean,
+    codegen: Boolean,
+    cost: Boolean): ExplainCommand = {
+    val sparkSession = SparkSession.getActiveSession
+    assert(sparkSession.nonEmpty, "Explain command shouldn't be initialized on executors")
+
+    val queryExecution =
+      if (logicalPlan.isStreaming) {
+        // This is used only by explaining `Dataset/DataFrame` created by `spark.readStream`, so the
+        // output mode does not matter since there is no `Sink`.
+        new IncrementalExecution(
+          sparkSession.get, logicalPlan, OutputMode.Append(), "<unknown>",
+          UUID.randomUUID, UUID.randomUUID, 0, OffsetSeqMetadata(0, 0))
+      } else {
+        sparkSession.get.sessionState.executePlan(logicalPlan)
+      }
+    new ExplainCommand(queryExecution, extended, codegen, cost)
+  }
+
+  /**
+   * This is mainly used for tests.
+   */
+  def apply(
```

Comment:
This is for test only. Because some test suites passes `logicalPlan` and `extended`, and we can't overload `apply` with default parameters. So to create this for test only purpose.